pkg/dataobj/planner/logical/table.go

```diff
@@ -0,0 +1,43 @@
+package logical
+
+import (
+	"github.com/grafana/loki/v3/pkg/dataobj/planner/schema"
+)
+
+// MakeTable represents a plan node that scans input data.
+// It is the leaf node in the query tree, representing the initial data source
+// from which all other operations will read. This is equivalent to a table scan
+// operation in a relational database.
+type MakeTable struct {
+	// name is the identifier of the table to scan
+	name string
+	// schema defines the structure of the data in the table
+	schema schema.Schema
+}
+
+// makeTable creates a new MakeTable plan node with the given name and schema.
+// This is an internal constructor used by the public NewScan function.
+func makeTable(name string, schema schema.Schema) *MakeTable {
+	return &MakeTable{
+		name:   name,
+		schema: schema,
+	}
+}
+
+// Schema returns the schema of the table.
+// This implements part of the Plan interface.
+func (t *MakeTable) Schema() schema.Schema {
+	return t.schema
+}
+
+// TableSchema returns the schema of the table.
+// This is a convenience method that returns the same value as Schema().
+func (t *MakeTable) TableSchema() schema.Schema {
+	return t.schema
+}
```

Comment:
This function is unused. Also, it's not a convenience method, but a duplicate ;-)